Userland/Libraries/LibGfx/ImageFormats/ImageDecoder.h

```diff
@@ -59,6 +60,7 @@ class ImageDecoder : public RefCounted<ImageDecoder> {
     bool is_animated() const { return m_plugin->is_animated(); }
     size_t loop_count() const { return m_plugin->loop_count(); }
     size_t frame_count() const { return m_plugin->frame_count(); }
+    size_t first_animated_frame_index() const { return m_plugin->first_animated_frame_index(); }
```

Comment:
Does this have to be part of the ImageDecoder api? Couldn't the png decoder handle this internally for animated pngs? (Like, it'd map 0 to the first animated index, etc)

Reply:
We could, but then how would you get the default image (e.g. for a thumbnail)?

Reply:
What if the thumbnail is just frame 0 (i.e. the first animated image)? Would that be worse than returning the default image?

Reply:
But what if you do want to see the "default" image (if it isn't part of the animation)? I don't know, I thought about this quite a bit and it seems like this is a pretty neat way to solve this issue:

If you want the (default) image, just call `decoder->frame(0)`. If you want to animate, just call `frame(auto i = decoder->first_animated_frame_index(); i < decoder->frame_count(); i++)`. Seems like a decent and straight forward way to do either? And then, depending on how the png is supposed to animate, this will automatically handle both cases correctly. If the default image should be part of the animation, this will include the default image in the animation, otherwise it will skip it.

Reply:
This also means that the thumbnail can always obtained with `frame(0)`.

Reply:
`decoder->frame(0)` not returning a frame of the animation doesn't seem super straightforward to me 🙃 It's also not good enough a thumbnail API, as eg jpegs can contain 5 different thumbnails.

it seems like it leaks apng implementation details into all the decoders.

(But just my opinion, which isn't worth more than yours. If you don't find this convincing, feel free to ignore!)

Reply:
Well if you want all 5 jpeg thumbnails you can just do:

`for (size_t i = 0; i < max(1, decoder->first_animated_frame_index()); i++)`

Anyway, I'm open to doing this differently, I'm open to suggestions.

Reply:
If we split `frames()` into separate functions like `thumbnails()`, `images()`, `animation_frames()` then all users will have to implement appropriate fall-back mechanisms. E.g. if no thumbail is present, then call `images()`, if that doesn't yield anything then maybe pick the first `animation_frames()` image.

Reply:
(note, not an argument: you can probably provide default implementation in the base case for all of the use cases as a fallback)

Reply:
Yeah, there's many ways. But ultimately, the ImageDecoder API will have to be enhanced one way or another. I'm open to suggestions.

Reply:
IMHO it's not something we need to change in this PR, is it? Just making frame(0) return the first anim frame for APNG and using that as thumbnail would be fine?

Reply:
I don't know, my guess would be that if the default image is not part of the animation then the first animation frame probably doesn't show any useful image?

Reply:
It's the first frame of the animation. Most animations loop. I think it'd often look ok.

Reply:
Unless it's e.g. a fade-in animation. In Ubuntu it uses the default image to generate a thumbnail. The first animation frame is entirely transparent.

![fade-in-error](https://user-images.githubusercontent.com/10320822/231002784-e2d5c8c4-5727-4349-af8c-2a2e27e15feb.png)

Reply:
(i feel that the solution for selecting the ‘previewʼ frame for animated pngs could be postponed to a separate pull request… 😅)

Reply:
What if we replace `frame(x)` with `image(x)`, `thumbnail(x)`, and `animation_frame(x)`?